ci-operator/step-registry/multiarch/nodes-readiness/multiarch-nodes-readiness-commands.sh

```diff
@@ -0,0 +1,115 @@
+    grep -c -E ",True$"
+}
+
+# wait_for_nodes_readiness loops until the number of ready nodes objects is equal to the desired one
+# It takes 3 arguments:
+# - expected_nodes: the number of nodes that should be ready
+# - max_retries: the maximum number of retries before failing. Default: 5.
+# - period: the time to wait between retries in minutes. Default: 1.
+function wait_for_nodes_readiness()
+{
+  local expected_nodes=${1}
+  local max_retries=${2:-20}
+  local period=${3:-1}
+  echo "[INFO] Waiting for ${expected_nodes} nodes to be ready..."
+  for i in $(seq 1 "${max_retries}") max; do
+    if [ "${i}" == "max" ]; then
+      echo "[ERROR] Timeout reached. ${expected_nodes} ready nodes expected, found ${ready_nodes}... Failing."
+      debug
+      return 1
+    fi
+    ready_nodes=$(get_ready_nodes_count)
+    if [ "${ready_nodes}" == "${expected_nodes}" ]; then
+        echo "[INFO] Found ${ready_nodes}/${expected_nodes} ready nodes, continuing..."
+        return 0
+    fi
+    echo "[INFO] - ${expected_nodes} ready nodes expected, found ${ready_nodes}..." \
+      "Waiting ${period}min before retrying (timeout in $(( (max_retries - i + 1) * (period) ))min)..."
+    sleep "${period}m"
+  done
+}
+
+# get_nodes_arch_by_label returns the architecture of the nodes with the given label.
+# If more architectures are detected, it returns space separated and sorted list of unique architectures.
+function get_nodes_arch_by_label() {
+  oc get nodes -l "${1}" \
+    -o jsonpath='{range .items[*]}{.metadata.name}{" "}{.metadata.labels.kubernetes\.io/arch}{"\n"}{end}' | \
+    cut -f2 -d' ' | sort -u | tr '\n' ' ' | sed -e 's/ $//'
+}
+
+# verify_arch_by_role_label checks if the nodes with the given label have the expected architecture
+# It takes 3 arguments:
+# - role: the role of the nodes to check. It's used for informational purposes only.
+# - label: the label to filter the nodes.
+# - expected: the expected architecture of the nodes.
+# The function depends on get_nodes_arch_by_label. Since it returns a sorted list of unique architectures,
+# this function can take a space separated list of architectures as the expected argument, for example,
+# to verify clusters with mixed architecture workers.
+function verify_arch_by_role_label() {
+  role="${1}"
+  label="${2}"
+  echo "[INFO] Checking the ${role} nodes architecture..."
+  expected="$(normalize_arch "${3}")"
+  ret_arch="$(normalize_arch "$(get_nodes_arch_by_label "${label}")")"
+  if ! [ "$ret_arch" == "${expected}" ]; then
+    echo "[ERROR] The expected ${role} architecture (${expected}) is different than the rendered one ($ret_arch)."
+    debug
+    return 1
+  fi
+  echo "[INFO] ${role} architecture verified ($ret_arch == ${expected})."
+}
+
+expected_nodes=$(( $(yq-v4 -r '.controlPlane.replicas // '"${CONTROL_PLANE_REPLICAS:-3}" "${SHARED_DIR}/install-config.yaml") ))
+if [ "${SIZE_VARIANT}" != "compact" ]; then
```

Comment:
Sounds like we do not need the if condition, when it is a compact cluster, compute replicas always need to set to "0", right?

Reply:
The idea is that this test should happen for both multi and non multi arch cases. In particular, the check for the readiness.. https://github.com/openshift/release/pull/54703#discussion_r1691288557

Maybe we can split this into two steps, I just believe it's good to move this readiness check right after the installation instead of expecting any further steps that come later to be responsible for ensuring it.

Reply:
> this test should happen for both multi and non multi arch cases.

If that, we should rename the step to `nodes-readiness` from `multiarch-nodes-readiness`.

And for this specific comment, I am suggesting to remove `[ "${SIZE_VARIANT}" != "compact" ]` if condition, because even it is a compact cluster, the following codes in this block still work, right?

Reply:
> If that, we should rename the step to nodes-readiness from multiarch-nodes-readiness.

Do we want one step (readiness + expected arch) or two separate steps?

Reply:
I prefer one step.

Reply:
damn, i just proposed two :D